hat/assets/js/apps/Iaso/domains/orgUnits/groups/components/Filters.tsx

```diff
@@ -0,0 +1,199 @@
+import React, {
+    FunctionComponent,
+    useState,
+    useMemo,
+    useCallback,
+} from 'react';
+import { Grid, Button } from '@mui/material';
+import SearchIcon from '@mui/icons-material/Search';
+import { useSafeIntl } from 'bluesquare-components';
+import { useGetProjectsDropdownOptions } from '../../../projects/hooks/requests';
+import { useDataSourceVersions } from '../../../dataSources/requests';
+import { useGetDataSources } from '../../../dataSources/useGetDataSources';
+import { useFilterState } from '../../../../hooks/useFilterState';
+import InputComponent from '../../../../components/forms/InputComponent';
+import { baseUrl } from '../config';
+import MESSAGES from '../messages';
+
+type Params = {
+    page?: string;
+    search?: string;
+    dataSource?: string;
+    version?: string;
+    project_ids?: string;
+    accountId: string;
+};
+
+type Props = {
+    params: Params;
+};
+
+const dataSourceByProjectId = (dataSources: any[], projectId: string) => {
+    const allDataSource: any[] = dataSources?.map(
+        (source: { projects: any[][]; name: any; id: any }) => {
+            const projectIds = source?.projects[0].map(
+                (project: { id: string }) => project?.id.toString(),
+            );
+            return {
+                label: source?.name,
+                value: `${source?.id}`,
+                projectIds,
+            };
+        },
+    );
+    if (projectId) {
+        return allDataSource?.filter((source: { projectIds: string[] }) =>
+            source.projectIds?.includes(projectId),
+        );
+    }
+    return allDataSource;
+};
+
+const versionsBySource = (sourceVersions: any[], source: string) => {
```

Comment:
This method can also be simplified.`const versionsBySource = (sourceVersions: any[], source: string) =>
    sourceVersions
        ?.filter(version => version?.data_source.toString() === source)
        ?.map(version => ({
            label: `${version?.data_source_name}-${version?.number}`,
            value: `${version?.id}`,
        }));`

Reply:
IMO, chaining filter and map functions would make code hard to read.